geps/overview.md

```diff
@@ -65,18 +65,53 @@ against them. At this stage, the GEP should be targeting the "Implementable"
 stage.
 
 ### 4. Implement the GEP as "Experimental"
+
+
 With the GEP marked as "Implementable", it is time to actually make those
 proposed changes in our API. In some cases, these changes will be documentation
 only, but in most cases, some API changes will also be required. It is important
 that every new feature of the API is marked as "Experimental" when it is
 introduced. Within the API, we use `<gateway:experimental>` tags to denote
 experimental fields.
 
-Before these changes are released, they will also need to be documented.
-GEPs that have not been both implemented and documented before a release
-cut off will be excluded from the release.
+Some other requirements must be met before marking a GEP `Experimental`:
+- the graduation criteria to reach `Standard` MUST be filled out
+- a proposed probationary period (see next section) must be included in the GEP
+  and approved by maintainers.
+
+Before changes are released they MUST be documented. GEPs that have not been
+both implemented and documented before a release cut off will be excluded from
+the release.
+
+### 5. Probationary Period
```

Comment:
I'm not sure where this comment belongs, but I'm a bit uncomfortable with the "Probationary" term here and adding another phase. I think what we may actually want here is just to clarify that:

* Experimental is a temporary phase
* GEPs will have regular check ins and must show progress towards standard channel
* As part of graduating to experimental, deadlines will be defined for key milestones on the path to GA, including final graduation
* If any of those deadlines are missed, we may choose to either:
  * Remove the GEP from the experimental channel (and therefore the API)
  * Provide a one-time extension of certain deadlines as agreed by the community

Reply:
I think I'm basically saying the same thing as you are, but the bits that were making me uncomfortable about the current wording were:

1. Is Probationary a phase in the GEP process? 
2. If so, do we now have two phases (Probationary and Experimental) that both land in experimental channel?
3. Does having a probationary state actually result in the "Experimental" state afterwards being viewed as more stable? IE once a GEP leaves probationary does it exist in Experimental channel without any risk of breaking changes or removal?

My suggestion above is just one possible way to address those questions, always open to other ideas.

Reply:
```
I think what we may actually want here is just to clarify that:

    Experimental is a temporary phase
    GEPs will have regular check ins and must show progress towards standard channel
    As part of graduating to experimental, deadlines will be defined for key milestones on the path to GA, including final graduation
    If any of those deadlines are missed, we may choose to either:
        Remove the GEP from the experimental channel (and therefore the API)
        Provide a one-time extension of certain deadlines as agreed by the community
```

I'm in a difficult spot in terms of how to progress because it occurs to me this is _exactly_ what's being covered with the current language?

> Is Probationary a phase in the GEP process?
> If so, do we now have two phases (Probationary and Experimental) that both land in experimental channel?
> Does having a probationary state actually result in the "Experimental" state afterwards being viewed as more stable?

"Probationary" as proposed here is part of the "GEP process" not the GEP phases (e.g. implementable, experimental, e.t.c.), in the same way that "Document Implementation Details" is not a phase, just to clarify that. The steps here do not all align with specific phases. Probation is being proposed as an _aspect_ of the `Experimental` phase.

---

In regards to your discomfort with the "probation" language I have the opposite feeling: I am strongly in favor of the language as I think it is accurate and cautionary and I like it front and center. Do you think you can be persuaded?

Reply:
After talking about it with Rob in slack, I think we have a compromise. Latest changes have merged the two "steps" together.